src/scripts/post_process_externally_managed_content.py

```diff
@@ -0,0 +1,112 @@
+    df_records = pd.DataFrame(report)
+    df_records.fillna("", inplace=True)
+    
+    if report:
+        report_string = df_records.to_markdown(index=False)
+    else:
+        report_string = "No QC failures found."
+    
+    failure_report = f"""
+# QC Report for {external}
+
+{report_string}
+"""
+    
+    with open(nice_report, "w") as f:
+        f.write(failure_report)
+
+def process_externally_managed_source(source, external_content_dir, robot_report_file):
+    df_robot_report = pd.read_csv(robot_report_file, sep="\t")
+    report = []
+    drop_rows_indexes = []
+    external_content_file = f"{external_content_dir}/{source}.robot.tsv"
+    external_content_file_out = f"{external_content_dir}/processed-{source}.robot.tsv"
+    if not os.path.exists(external_content_file):
+        logger.warning(f"External content file {external_content_file} does not exist.")
+        return
+    df_external_content = pd.read_csv(external_content_file, sep="\t")
+    for _, qc_failure in df_robot_report.iterrows():
+        mondo_id_failure = qc_failure["Subject"]
+        if mondo_id_failure in df_external_content.iloc[:, 0][1:].tolist():
+            erroneous_rows = df_external_content[df_external_content.iloc[:, 0] == mondo_id_failure]
+            for index, erroneous_row in erroneous_rows.iterrows():
+                ## if the row is erroneous because of qc_failure is related to this specififc external content
+                erroneous_column = is_qc_failure_related_to_external_source(qc_failure, erroneous_row, source)
+                if erroneous_column:
+                    erroneous_row_cp = erroneous_row.copy()
+                    for col in erroneous_rows.columns:
+                            if col != erroneous_rows.columns[0] and col != erroneous_column:
+                                erroneous_row_cp.at[col] = ""
+                    error_report = erroneous_row_cp.to_dict()
+                    error_report['Source'] = source
+                    rule = qc_failure["Rule Name"]
+                    property = qc_failure["Property"]
+                    error_report['Check'] = f"{rule} ({property})"
+                    report.append(error_report)
+                    df_external_content.at[index, erroneous_column] = ""
+
+    df_external_content.drop(drop_rows_indexes, inplace=True)
+    df_external_content.to_csv(external_content_file_out, sep="\t", index=False)
+    write_nice_report(report, source)
+
+
+@click.command()
+@click.option('--emc-id', help='ID of the externally managed content file to process.', required=True, multiple=True)
+@click.option('--emc-dir', help='Directory where the externally managed content (emc) is located.')
+@click.option('--robot-report', help='Location of the robot report with violations.')
+def post_process_externally_managed_content(emc_id, emc_dir, robot_report):
+    """
+    Post-process externally managed content.
+    """
+    logger.error("Post-processing externally managed content.")
```

Comment:
These look like they should be 'info'. Is `logger.error()` temporary?

Reply:
Thanks for pointing that out, fixed.